pkg/aspect/quotasManager_test.go

```diff
@@ -14,8 +14,319 @@
+	return 0, nil
+}
+
+type fakeQuotaBuilder struct {
+	adapter.Builder
+	name string
+
+	body func() (adapter.QuotasAspect, error)
+}
+
+func (b *fakeQuotaBuilder) Name() string {
+	return b.name
+}
+
+func (b *fakeQuotaBuilder) NewQuotasAspect(env adapter.Env, config adapter.AspectConfig,
+	quotas map[string]*adapter.QuotaDefinition) (adapter.QuotasAspect, error) {
+	return b.body()
+}
+
+func TestNewQuotasManager(t *testing.T) {
+	m := NewQuotasManager()
+	if m.Kind() != QuotasKind {
+		t.Errorf("m.Kind() = %s wanted %s", m.Kind(), QuotasKind)
+	}
+	if err := m.ValidateConfig(m.DefaultConfig()); err != nil {
+		t.Errorf("m.ValidateConfig(m.DefaultConfig()) = %v; wanted no err", err)
+	}
+}
+
+func TestQuotasManager_NewAspect(t *testing.T) {
+	builder := &fakeQuotaBuilder{name: "test", body: func() (adapter.QuotasAspect, error) {
+		return &fakeQuotaAspect{}, nil
+	}}
+
+	conf := &config.Combined{
+		Aspect: &pb.Aspect{
+			Params: &aconfig.QuotasParams{
+				Quotas: []*aconfig.QuotasParams_Quota{
+					{
+						Descriptor_: "RequestCount",
+						Labels:      map[string]string{"source": "", "target": ""},
+					},
+				},
+			},
+		},
+		// the params we use here don't matter because we're faking the aspect
+		Builder: &pb.Adapter{Params: &aconfig.QuotasParams{}},
+	}
+	if _, err := NewQuotasManager().NewAspect(conf, builder, atest.NewEnv(t)); err != nil {
+		t.Errorf("NewAspect(conf, builder, test.NewEnv(t)) = _, %v; wanted no err", err)
+	}
+
+	conf = &config.Combined{
+		Aspect: &pb.Aspect{
+			Params: &aconfig.QuotasParams{
+				Quotas: []*aconfig.QuotasParams_Quota{
+					{
+						Descriptor_: "FOOBAR",
+						Labels:      map[string]string{"source": "", "target": ""},
+					},
+				},
+			},
+		},
+		// the params we use here don't matter because we're faking the aspect
+		Builder: &pb.Adapter{Params: &aconfig.QuotasParams{}},
+	}
+	if _, err := NewQuotasManager().NewAspect(conf, builder, atest.NewEnv(t)); err != nil {
+		t.Errorf("NewAspect(conf, builder, test.NewEnv(t)) = _, %v; wanted no err", err)
+	}
+}
+
+func TestQuotasManager_NewAspect_PropagatesError(t *testing.T) {
+	conf := &config.Combined{
+		Aspect: &pb.Aspect{Params: &aconfig.QuotasParams{}},
+		// the params we use here don't matter because we're faking the aspect
+		Builder: &pb.Adapter{Params: &aconfig.QuotasParams{}},
+	}
+	errString := "expected"
+	builder := &fakeQuotaBuilder{
+		body: func() (adapter.QuotasAspect, error) {
+			return nil, fmt.Errorf(errString)
+		}}
+	_, err := NewQuotasManager().NewAspect(conf, builder, atest.NewEnv(t))
+	if err == nil {
+		t.Error("NewQuotasManager().NewAspect(conf, builder, test.NewEnv(t)) = _, nil; wanted err")
+	}
+	if !strings.Contains(err.Error(), errString) {
+		t.Errorf("NewAspect(conf, builder, test.NewEnv(t)) = _, %v; wanted err %s", err, errString)
+	}
+}
+
+func TestQuotaWrapper_Execute(t *testing.T) {
+	// TODO: all of these test values are hardcoded to match the metric definitions hardcoded in metricsManager
```

Comment:
wait, why are quota tests depending on the hardcoded values of the metrics stuff?